proxy/src/schema/registry.rs

```diff
@@ -0,0 +1,88 @@
+use futures::future::Future;
+
+use radicle_registry_client::{
+    ed25519::Pair, ClientT, CreateCheckpointParams, RegisterProjectParams, String32, H256,
+};
+
+use crate::schema::error::{Error, ProjectValidation};
+
+/// Registry client wrapper.
+#[derive(Clone)]
+pub struct Registry<R>
+where
+    R: ClientT,
+{
+    /// Registry client, whether an emulator or otherwise.
+    client: R,
```

Comment:
It’s not necessary to make this polymorphic in the client. We changed this recently and now there exits only one implementation of client which is `Client`. So you could remove the type parameter `R` every where and replace it with `Client`.

Of course you might only want to rely on methods defined by `ClientT` in which case you can keep it like it is.

Reply:
This causes some confusing issues so I stuck with ClientT.